<launch>

    <arg name="imu_type" default="mpu6000"/>
    <arg name="slam_sensor_type" default="laser"/> <!-- stereo, laser -->
    <arg name="slam_pose_topic" default="/slam/pose"/>
    <arg name="source_msg_parent_frame" default="base_footprint"/>
    <arg name="source_msg_child_frame" default="gyro_link"/>
 
    <param name="/use_sim_time" value="true" />

    <!-- call msf for visual-inertial state estimation -->
    <node name="msf_pose_sensor" pkg="msf_updates" type="pose_sensor" clear_params="true" output="screen">
        <remap from="msf_core/imu_state_input" to="/imu0" />
        <remap from="msf_updates/pose_with_covariance_input" to="$(arg slam_pose_topic)" />
        <rosparam file="$(find msf_updates)/gazebo_$(arg slam_sensor_type)_$(arg imu_type)_fix.yaml"/>
        <!-- <remap from="/msf_core/odometry" to="/odom" /> -->
    </node>

    <!-- @TODO convert msf to odom and pub map_to_odom tf -->

    <!-- initialize msf -->
    <node pkg="rosservice" type="rosservice" name="initialize" args="call --wait /msf_pose_sensor/pose_sensor/initialize_msf_scale 1"/>

</launch>
  
